### PR TITLE
fix(engine): return original val if not a string for string_to_tag_array

### LINKED ISSUE #7391: string_to_tag_array() should not return false if array is passed
Just noticed that passing an array to ```string_to_tag_array()``` results in false. Perhaps we should spit that array right back out.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #7391
- [x] add test

### CODE REVIEW

File: engine/lib/tags.php [hunk 1]
```diff
@@ -13,9 +13,9 @@
  *
  * @param string $string Comma-separated tag string
  *
- * @return array|false An array of strings, or false on failure
+ * @return array|false|mixed An array of strings, false if no data provided, or the original data if input was not a string
  */
-function string_to_tag_array($string) {
+function string_to_tag_array($string = false) {
```
Comment (mrclay) [inline engine/lib/tags.php]: We should not make the argument optional, but I think we should rename it "tags". It's confusing to see a `return $string` for the case when we know it's not a string :)

**Comment by mrclay** [comment]
A phpunit test would be easy for this.

File: engine/lib/tags.php (hunk 1 above)
Comment (jdalsem) [inline engine/lib/tags.php]: why not call it `$input` as it is, potentially, also not `tags`. Then we can say `return $input;`

Comment (jdalsem) [inline engine/lib/tags.php]: And how do we need the function doc? Should it say: 
`@param mixed $input comma-separated string` ??

Comment (mrclay) [inline engine/lib/tags.php]: $input is fine. Pretty much anything except $string ;) I think we leave the `@param` as string. This is what the function is meant for, we're just handling idiosyncrasies in how it's commonly used.

Comment (mrclay) [inline engine/lib/tags.php]: In fact I think I'm wrong to suggest changing the var name. It should be $string, but we should handle the invalid input first:
```php
if (!is_string($string)) {
    return $string;
}
// do the split...
```

**Pull request merged**